### ISSUE TITLE
Stroke Recognizer makes straight lines different than expected 

### ISSUE THREAD

**Issue by c-ola** [xournalpp/xournalpp#5637]
### Operating System

Linux

### (Linux only) Distribution

Fedora

### (Linux only) Desktop Environment

i3

### (Linux Only) Display Server

X.Org

### Installation Method

dnf package manager

### Xournal++ Version

1.2.3

### libgtk Version

3.24.41

### Bug Description

When using the stroke recognizer to draw straight diagonal lines, the line will sometimes get shortened if you go back towards the direction of the starting point (or even in the opposite direction if you go over the starting point).

### Expected Behaviour

I would expect the line to be straight along where it is drawn (go from min to max).

### Steps to Reproduce

1. Start drawing line diagonally.
2. Go back in direction of start point
3. Then release the drawing tool that you are using (occurs with both pen and highlighter

### Additional Context

 It also does not happen when the line is parallel to either the x or y axes.

 Drawing a line in the following way (before releasing the drawing tool):
![image](https://github.com/xournalpp/xournalpp/assets/59809814/c4ee643f-309f-4bb1-b438-06604cb23cf2)
Leads to this image:
![image](https://github.com/xournalpp/xournalpp/assets/59809814/8a3223a6-43df-4dbc-81d8-cd436797a505)

This is an example of it not happening parallel to the x axis.
![image](https://github.com/xournalpp/xournalpp/assets/59809814/3f438c98-0379-4ebc-9d0a-efd610ac8462)
![image](https://github.com/xournalpp/xournalpp/assets/59809814/989a473d-d5fe-4075-84a0-84a383c5ca57)

**Comment by Febbe** [comment]
@kenballus you implemented the preserve the endpoint preservation in #3285, can you tell me, if you introduced the bug there?

**Comment by kenballus** [comment]
Yes; I introduced this behavior. This is how I prefer the stroke recognizer to behave, but that is ultimately just an opinion. Keep in mind that my usual use case for the stroke recognizer is drawing straight lines between preexisting points on a page, so it is in my interested to precisely control the start and end points of a recognized line segment.

**Comment by c-ola** [comment]
In that sense the behaviour makes sense, however, shouldn't it then behave the same way along the x and y axes as well?

**Comment by kenballus** [comment]
Oh, sorry. I misunderstood the issue here.

I am not sure whether I introduced this behavior. If I did, it was definitely unintentional, and should be fixed.

**Comment by lluisalemanypuig** [comment]
I have recorded [a video](https://youtu.be/DkZyGt-G-oA) showing the issues pointed out by @c-ola. I know how to fix this, but I would have to remove code that is supposed to fix other bugs.
